libp2p/peer/peerstore.py

```diff
@@ -10,90 +12,196 @@
 
 class PeerStore(IPeerStore):
 
-    peer_map: Dict[ID, PeerData]
+    peer_pubkey_map: Dict[ID, PublicKey]
+    peer_privkey_map: Dict[ID, PrivateKey]
+    peer_data_map: Dict[ID, PeerData]
```

Comment:
we are probably following the Go implementation on this but does it make sense to attach the pubkey/privkey to the `PeerData` directly?

so then you just have one map from `ID` to `PeerData` and then those fields are (optionally?) sitting on the instance of `PeerData`.

the general thinking here is the fewer places to synchronize the better.

Reply:
Good idea!